### PR TITLE
integ-tests: use egoscale v3

### PR DESCRIPTION
**Pull request by sauterp**
# Description
We remove egoscale v2 from the integration tests(at the time v3 wasn't available). This will enable testing whether the behavior of storage class retain policies is correct.

## Checklist
(For exoscale contributors)

* [x] Changelog updated (under *Unreleased* block)
* [x] Integration tests OK

### CODE REVIEW

File: internal/integ/cluster/utils.go [hunk 1]
```diff
@@ -75,40 +75,50 @@ func getCredentialsFromEnv() (string, string, error) {
...
 }
 
-func (c *Cluster) getClusterID() (string, error) {
+func (c *Cluster) getClusterID() (exov3.UUID, error) {
 	if err := flags.ValidateFlags(); err != nil {
 		return "", err
 	}
 
-	cluster, err := c.Ego.FindSKSCluster(c.exoV2Context, *flags.Zone, *flags.ClusterName)
+	clusterList, err := c.Ego.ListSKSClusters(c.context)
```
Comment (pierre-emmanuelJ) [inline internal/integ/cluster/utils.go:83]: clusterList should have a method find, or you need to bump Egoscale to latest :)

Comment (pierre-emmanuelJ) [reply internal/integ/cluster/utils.go:83]: Just checked, latest version of egoscale have it
```Golang
FindSKSCluster(nameOrID string) (v3.SKSCluster, error)

FindSKSCluster attempts to find an SKSCluster by name or ID.
```

File: internal/integ/cluster/utils.go [hunk 2]
```diff
@@ -54,214 +51,141 @@ func ptr[T any](v T) *T {
...
 			},
 		}
 
-		role, err := c.Ego.CreateIAMRole(c.exoV2Context, *flags.Zone, &exov2.IAMRole{
-			Name:        ptr(c.APIRoleName),
-			Description: ptr("role for the CSI test cluster " + c.Name),
+		op, err := c.Ego.CreateIAMRole(ctx, exov3.CreateIAMRoleRequest{
+			Name:        c.APIRoleName,
+			Description: "role for the CSI test cluster " + c.Name,
 			Editable:    ptr(false),
```
Comment (pierre-emmanuelJ) [inline internal/integ/cluster/utils.go:163]: Now it's possible to use built-in `exov3.Ptr()`

File: internal/integ/cluster/utils.go [hunk 3]
```diff
@@ -54,214 +51,141 @@ func ptr[T any](v T) *T {
...
 		if err != nil {
 			return fmt.Errorf("error creating IAM role: %w", err)
 		}
 
-		apikey := &exov2.APIKey{
-			Name:   ptr(c.APIKeyName),
-			RoleID: role.ID,
-		}
+		// necessary due to issue [sc-91482]
+		time.Sleep(3 * time.Second)
```
Comment (pierre-emmanuelJ) [inline internal/integ/cluster/utils.go:178]: Thanks for the bug report in the API

**Review by pierre-emmanuelJ — APPROVED** [review]

**Review by kobajagi — APPROVED** [review]

**Pull request merged**